api/v1beta1/helmchart_types.go

```diff
@@ -50,6 +50,13 @@ type HelmChartSpec struct {
 	// +optional
 	ValuesFile string `json:"valuesFile,omitempty"`
 
+	// Ignore the version of the chart, allowing reconciliation if the source has
+	// changed. Only applies to GitRepository and Bucket sources. Defaults to false
+	// when omitted.
+	// +kubebuilder:default:=false
+	// +optional
+	IgnoreVersion bool `json:"ignoreVersion,omitempty"`
```

Comment:
I think this is a horrible name and doesn't convey what it does well.

Reply:
`SourceRevision`
> Use the revision of the Helm chart source instead of the version in the Chart.yaml. Only applicable to `GitRepository` and `Bucket` sources.

Reply:
`RevisionFromSource`

Reply:
Having thought a bit more about the usefulness about this feature, I think it may be a great idea to _always_ add the build metadata to the SemVer that is created for the chart, but only take it into account if some flag is enabled.

That will make the naming of this flag even harder.

Reply:
`ReconcileFromSource`? In that case I think `Reconcile` based on some condition is the right way to phrase it.

Reply:
But it _always_ reconciles from a source, the conditions on when it happens can only be adjusted which makes it more like a policy.

We could introduce a `<something>Policy` that can either be `ChartVersion` or `SourceRevision` to make things more descriptive, as [done here](https://github.com/alex-berger/helm-controller/blob/feature/crd-upgrade/api/v2beta1/helmrelease_types.go#L424-L435), but the `<something>` would then still require a name.

Reply:
`ReconcilePolicy`? 😅

Reply:
I agree that it should be an enum instead of a boolean and that it end in something like Policy or Strategy. I like `ReconciliationPolicy`.

Reply:
`ReconciliationPolicy` is fine with me, I think the description should be a bit more descriptive, like:

> ReconcilePolicy defines the reconciliation approach used when determining whether a Source has changed and a new Artifact must be produced.

Reply:
Given English is not my mother tongue, I do however wonder if it shouldn't be `ReconcileStrategy`, as it defines what to do _during_ reconciliation, and not _when to_ reconcile.

Reply:
English is my only language but I made sure to avoid most grammar lessons.

I prefer `Reconcile` to `Reconciliation` because it's simpler.

Reply:
Some prior (albeit Java) art: https://docs.oracle.com/cd/E10926_01/doc/owb.101/b12155/oracle/owb/reconcile/ReconcileStrategy.html